### PR TITLE
Make CI profile configurable via env vars

### PR DESCRIPTION
**Pull request by fanny-jiang**
### What does this PR do?

Makes the CI profile creation configurable using env vars: `CI_ENV_NAMES` to configure an AWS environment name other than `aws/agent-qa` and `CI_SECRET_PREFIX` to configure a secrets prefix other than `ci.datadog-agent.`. 

### Motivation

The changes will enable the `runner` and `infra` packages to be used in external repos to run pulumi-based E2E tests in CI. Currently, the `runner` module is hardcoded to use `datadog-agent`-specific CI secrets. In order to use the module in an external repo, some unexported methods need to be patched and this change avoids that need. 

### Describe how to test/QA your changes

No QA/run E2E tests in CI to confirm that the default values are still being used if the `CI_ENV_NAMES` and `CI_SECRET_PREFIX` env vars are not set. 

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/runner/profile.go [hunk 1]
```diff
@@ -17,10 +17,11 @@ import (
...
 const (
-	AWS       CloudProvider = "aws"
-	Azure     CloudProvider = "az"
-	GCP       CloudProvider = "gcp"
-	EnvPrefix               = "E2E_"
+	AWS                   CloudProvider = "aws"
+	Azure                 CloudProvider = "az"
+	GCP                   CloudProvider = "gcp"
+	EnvPrefix                           = "E2E_"
+	DefaultCISecretPrefix               = "ci.datadog-agent."
```
Comment (pducolin) [inline test/new-e2e/runner/profile.go:24]: **💬 suggestion**
As this is CI related, would define it in `ci-profile.go` instead

Comment (pducolin) [reply test/new-e2e/runner/profile.go:24]: Same for `defaultCIEnvs`

**Review by pducolin — COMMENTED** [review]
Minor change, other than that makes sense to have these changes

**Review by ogaca-dd — COMMENTED** [review]
LGTM with a fix for https://github.com/DataDog/datadog-agent/pull/17003/files#r1188283892

**Review by ogaca-dd — APPROVED** [review]

**Review by pducolin — APPROVED** [review]

**Review by ogaca-dd — APPROVED** [review]

File: test/new-e2e/runner/ci-profile.go [hunk 2]
```diff
@@ -8,10 +8,15 @@ package runner
 import (
 	"fmt"
 	"os"
+	"strings"
 
 	"github.com/DataDog/datadog-agent/test/new-e2e/runner/parameters"
 )
 
+const DefaultCISecretPrefix = "ci.datadog-agent."
```
Comment (vboulineau) [inline test/new-e2e/runner/ci-profile.go:16]: Should be private

**Pull request merged**